File: core/src/main/scala/com/graphbrain/eco/nodes/PosFun.scala
package com.graphbrain.eco.nodes

import com.graphbrain.eco.NodeType.NodeType
import com.graphbrain.eco.{Contexts, NodeType}

class PosFun(params: Array[ProgNode], lastTokenPos: Int= -1) extends FunNode(params, lastTokenPos) {
  override val label = "pos"

  override def ntype: NodeType = NodeType.Number

  override def numberValue(ctxts: Contexts) = {
    val p = params(0)
    p.wordsValue(ctxts)
    for (c <- ctxts.ctxts) {
      val len = c.getRetWords(p).count
      c.setRetNumber(this, len)
    }
  }
}

File: core/src/main/java/com/graphbrain/eco/nodes/PosFun.java
package com.graphbrain.eco.nodes;

import com.graphbrain.eco.Context;
import com.graphbrain.eco.NodeType;
import com.graphbrain.eco.Contexts;

public class PosFun extends FunNode {

    public PosFun(ProgNode[] params, int lastTokenPos) {
        super(params, lastTokenPos);
    }

    public PosFun(ProgNode[] params) {
        this(params, -1);
    }

    @Override
    public String label(){return "pos";}

    @Override
    public NodeType ntype(){return NodeType.Number;}

    @Override
    public void numberValue(Contexts ctxts) {
        ProgNode p = params[0];
        p.wordsValue(ctxts);
        for (Context c : ctxts.getCtxts()) {
            double pos = c.getRetWords(p).getPos();
            c.setRetNumber(this, pos);
        }
    }
}